src/gpb_compile.erl

```diff
@@ -1820,6 +1820,9 @@ format_erl(Mod, Defs, #anres{maps_as_msgs=MapsAsMsgs}=AnRes, Opts) ->
        %% to about 10000 msgs/s for a set of mixed message samples.
        %% f("-compile(inline).~n"),
        %%
+       %% Add epb compatibility function
+       ?f("-spec encode(_) -> no_return().~n"),
+       ?f("encode(Msg) -> encode_msg(Msg, []).~n"),
```

Comment:
> +       ?f("-spec encode(_) -> no_return().~n"),
> +       ?f("encode(Msg) -> encode_msg(Msg, []).~n"),

The spec looks odd. Only if there are no messages defined in the `.proto` file is the spec valid, but for normal cases, something is actually returned. You wrote that _with same signature as what is used in erlang_protobuffs_, I don't know exactly what that means, but I think it would be good to be more accurate: emit the `-spec...no_return()` only if there are no messages; the [`gpb_compile:contains_messages/1` function](https://github.com/tomas-abrahamsson/gpb/blob/3.23.2/src/gpb_compile.erl#L7692) can be useful here.